fake/fake-extension.js

```diff
@@ -217,9 +217,9 @@ var seedMvcActions = (function() {
                             { access: 'middleware-start', correlationId: '3', name: 'logger', displayName: 'Morgan Logger', packageName: 'morgan', paths: ['/'] },
                             { access: 'middleware-end', correlationId: '3', name: 'logger', displayName: 'Morgan Logger', packageName: 'morgan', paths: ['/'], result: 'next' },
                             { access: 'middleware-start', correlationId: '4', name: 'jsonParser', displayName: 'JSON Body Parser', packageName: 'body-parser', paths: ['/'] },
-                            { access: 'middleware-end', correlationId: '4', name: 'jsonParser', displayName: 'JSON Body Parser', packageName: 'body-parser', paths: ['/'], result: 'next' },
+                            { access: 'middleware-end', correlationId: '4', name: 'jsonParser', displayName: 'JSON Body Parser', packageName: 'body-parser', paths: ['/'], headers: [{ op: 'set', name: 'content-type', value: 'application/json'}], result: 'next' },
                             { access: 'middleware-start', correlationId: '5', name: 'urlencodedParser', displayName: 'URL-Encoded Body Parser', packageName: 'body-parser', paths: ['/'] },
-                            { access: 'middleware-end', correlationId: '5', name: 'urlencodedParser', displayName: 'URL-Encoded Body Parser', packageName: 'body-parser', paths: ['/'], result: 'next' },
+                            { access: 'middleware-end', correlationId: '5', name: 'urlencodedParser', displayName: 'URL-Encoded Body Parser', packageName: 'body-parser', paths: ['/'], headers: [{ op: 'set', name: 'content-type', value: 'application/xml'}], result: 'next' },
```

Comment:
I'd be a little concerned with the message schema becoming a little too "meta" and loose in that kind of an attempt to be able to describe every kind of change that might occur during middleware execution.  I think we'll still want to have concept specific sub-schemas for noting changes to specific objects (e.g. headers, status codes, body, cookies, etc.).